test/export/FHIRExporter.test.ts

```diff
@@ -175,6 +175,266 @@ describe('FHIRExporter', () => {
+      const instance = new Instance('MyValueSet');
+      instance.instanceOf = 'ValueSet';
+      instance.usage = 'Definition';
+      const vsName = new AssignmentRule('name');
+      vsName.value = 'MyVS';
+      instance.rules.push(vsName);
+      doc.instances.set(instance.name, instance);
+
+      const profile = new Profile('ContainingProfile');
+      profile.parent = 'Basic';
+      const caretValueRule = new CaretValueRule('');
+      caretValueRule.caretPath = 'contained';
+      caretValueRule.value = 'MyValueSet';
+      caretValueRule.isInstance = true;
+      const bindingRule = new BindingRule('code');
+      bindingRule.valueSet = 'MyVS';
+      bindingRule.strength = 'extensible';
+      profile.rules.push(caretValueRule, bindingRule);
+      doc.profiles.set(profile.name, profile);
+
+      const result = exporter.export();
+      expect(result.profiles).toHaveLength(1);
+      expect(result.profiles[0].contained).toEqual([
+        {
+          resourceType: 'ValueSet',
+          id: 'MyValueSet',
+          url: 'http://hl7.org/fhir/us/minimal/ValueSet/MyValueSet',
+          name: 'MyVS'
+        }
+      ]);
+      const codeElement = result.profiles[0].findElement('Basic.code');
+      expect(codeElement.binding.strength).toBe('extensible');
+      expect(codeElement.binding.valueSet).toBe('#MyValueSet');
+    });
+
+    it('should allow a profile to bind an element to a contained ValueSet using a relative reference when the rule includes a version', () => {
+      const valueSet = new FshValueSet('MyValueSet');
+      valueSet.id = 'my-value-set';
+      const vsVersion = new CaretValueRule('');
+      vsVersion.caretPath = 'version';
+      vsVersion.value = '1.2.8';
+      valueSet.rules.push(vsVersion);
+      doc.valueSets.set(valueSet.name, valueSet);
+
+      const profile = new Profile('ContainingProfile');
+      profile.parent = 'Basic';
+      const caretValueRule = new CaretValueRule('');
+      caretValueRule.caretPath = 'contained';
+      caretValueRule.value = 'MyValueSet|1.2.8';
+      caretValueRule.isInstance = true;
+      const bindingRule = new BindingRule('code');
+      bindingRule.valueSet = 'MyValueSet';
+      bindingRule.strength = 'extensible';
+      profile.rules.push(caretValueRule, bindingRule);
+      doc.profiles.set(profile.name, profile);
+
+      const result = exporter.export();
+      expect(result.profiles).toHaveLength(1);
+      expect(result.profiles[0].contained).toEqual([
+        {
+          resourceType: 'ValueSet',
+          id: 'my-value-set',
+          name: 'MyValueSet',
+          url: 'http://hl7.org/fhir/us/minimal/ValueSet/my-value-set',
+          status: 'draft',
+          version: '1.2.8'
+        }
+      ]);
+      const codeElement = result.profiles[0].findElement('Basic.code');
+      expect(codeElement.binding.strength).toBe('extensible');
+      expect(codeElement.binding.valueSet).toBe('#my-value-set');
```

Comment:
Is there somewhere in the spec that says the fragment references do not specify version?

Reply:
Based on what I'm seeing in [the docs for contained resources](https://hl7.org/fhir/R5/references.html#contained), I think that version does not get included here. Specifically:
> Regarding history, it's possible to point to a contained resource within a version of the containing resource (via `[type]/[id]/_history/[version]#[containedId]`), but it's not possible to refer to a version of a contained resource (i.e. this is invalid: `[type]/[id]#[containedId]/_history/[version]`)

Reply:
Hm I'm not sure I know much about the `_history` piece, but I think from reading that section, I don't see any mention of including the version directly, so I think you're right that this is set. If we learn more later, we can always adjust later on and this test is a good demonstration of the current behavior.